src/js/stores/DCOSStore.js

```diff
@@ -130,20 +149,19 @@ class DCOSStore extends EventEmitter {
    * @type {ServiceTree}
    */
   get serviceTree() {
-    let {marathon, mesos} = this.data;
+    let {marathon:{tree}, mesos:{frameworks}} = this.data;
 
     // Merge Mesos and Marathon data
-    if (mesos.frameworks.length > 0) {
+    if (frameworks.length > 0) {
       // Create framework dict from Mesos data
-      let frameworks = mesos.frameworks
-        .reduce(function (map, framework) {
-          map[framework.name] = framework;
+      frameworks = frameworks.reduce(function (map, framework) {
+        map[framework.name] = framework;
 
-          return map;
-        }, {});
+        return map;
```

Comment:
This is not a `Map` – so naming it `map` is kinda confusing. How about the classic `memo`?